src/app/loadout-builder/process/useProcess.ts

```diff
@@ -222,47 +228,60 @@ const groupComparator = chainComparator(
 );
 
 /**
- * This groups items for process depending on whether any general, other or raid mods are locked as follows
- * - If there are general, other or raid mods locked it groups items by (stats, masterworked, modSlot, energyType).
- * - If there are only general mods locked it groupes items by (stats, masterwork, energyType)
- * - If no general, other or raid mods are locked it groups by (stats, masterworked).
+ * To reduce the number of items sent to the web worker we group items by a number of varying
+ * parameters, depending on what mods and armour upgrades are selected.
+ *
+ * After items have been grouped we only send a single item (the first one) as a representative of
+ * said group. All other grouped items will be available by the swap icon in the UI.
  *
- * Note that assumedMasterwork effects this.
+ * It can group by any number of the following concepts depending on locked mods and armor upgrades,
+ * - Stat distribution
+ * - Masterwork status
+ * - If there are energy requirements for slot independent mods it creates groups split by energy type
+ * - If there are activity mods it will create groups split by specialty socket tag
  */
 function groupItems(
   defs: D2ManifestDefinitions | undefined,
   items: readonly DimItem[],
   statOrder: number[],
   upgradeSpendTier: UpgradeSpendTier,
+  lockItemEnergyType: boolean,
   generalMods: PluggableInventoryItemDefinition[],
-  raidCombatAndLegacyMods: PluggableInventoryItemDefinition[]
+  combatMods: PluggableInventoryItemDefinition[],
+  activityMods: PluggableInventoryItemDefinition[]
 ) {
   const groupingFn = (item: DimItem) => {
     const statValues: number[] = [];
     const statsByHash = item.stats && keyByStatHash(item.stats);
     // Ensure ordering of stats
     if (statsByHash) {
       for (const statHash of statOrder) {
-        statValues.push(statsByHash[statHash]!.base);
+        let value = statsByHash[statHash]!.base;
+        if (defs && upgradeSpendTierToMaxEnergy(defs, upgradeSpendTier, item) === 10) {
+          value += 2;
+        }
+        statValues.push(value);
       }
     }
 
-    let groupId = `${statValues}${
-      defs && upgradeSpendTierToMaxEnergy(defs, upgradeSpendTier, item) === 10
-    }`;
+    let groupId = `${statValues}`;
```

Comment:
I deleted the boolean masterwork status and actually calculated the stats with masterwork status accounted for. While the last one did the job for class items specifically, this is just more accurate and makes more sense.